src/solver/conv_asm_implicit_gemm_v4r1_dynamic.cpp

```diff
@@ -0,0 +1,607 @@
+    // as vector_type<vector<half,4>, 2> is not working. So, restrict epack*gemmreada <= 4
+    // and epack*gemmreadb <= 4
+    // if((ctx.IsFp16()  || ctx.IsBfp16()) && ((GetEPackLength(ctx, false)*GemmNPerThreadSubC > 4)
+    // ||
+    //   (GetEPackLength(ctx, false)*GemmMPerThreadSubC > 4)))
+    //  return false;
+
+    // sanity check
+    if((KPerBlock % (GemmMPerThreadSubC * GemmMLevel0Cluster * GemmMLevel1Cluster)) != 0)
+        return false;
+
+    if(GemmNRepeat !=
+       (N1 * N2 * BPerBlock) / (GemmNPerThreadSubC * GemmNLevel0Cluster * GemmNLevel1Cluster))
+        return false;
+
+    const int ThreadPerLevel1Cluster =
+        GemmMLevel0Cluster * GemmNLevel0Cluster * GemmMLevel1Cluster * GemmNLevel1Cluster;
+
+    const int block_size = ThreadPerLevel1Cluster;
+
+    if(block_size < 64 || block_size > 512)
+        return false;
+
+    if(block_size !=
+       InBlockCopyClusterLengths_E * InBlockCopyClusterLengths_N1 * InBlockCopyClusterLengths_B *
+           InBlockCopyClusterLengths_N2)
+        return false;
+
+    if(block_size != WeiBlockCopyClusterLengths_K * WeiBlockCopyClusterLengths_E)
+        return false;
+
+    const int GemmMRepeat =
+        KPerBlock / (GemmMPerThreadSubC * GemmMLevel0Cluster * GemmMLevel1Cluster);
+
+    if(!(GemmMRepeat == 2 && GemmNRepeat == 2))
+        return false;
+
+    const int InBlockCopySubLengths_E  = EPerBlock / InBlockCopyClusterLengths_E;
+    const int InBlockCopySubLengths_B  = BPerBlock / InBlockCopyClusterLengths_B;
+    const int WeiBlockCopySubLengths_K = KPerBlock / WeiBlockCopyClusterLengths_K;
+
+    const std::size_t lds_size = ComputeLDSRequiredSize(ctx,
+                                                        BPerBlock,
+                                                        KPerBlock,
+                                                        EPerBlock,
+                                                        GemmMPerThreadSubC,
+                                                        GemmNPerThreadSubC,
+                                                        InBlockCopySubLengths_B,
+                                                        WeiBlockCopySubLengths_K,
+                                                        GetEPackLength(ctx, false));
+
+    if(lds_size > 64 * 1024)
+        return false;
+
+    return (InBlockCopySubLengths_E == 1 && InBlockCopySubLengths_B == 1);
+}
+
+void PerformanceImplicitGemmV4R1Dynamic::EuristicInit(const ConvolutionContext& config)
+{
+    auto tunables = GetImplicitGemmV4R1DynamicTunables();
+    auto it       = std::find_if(
+        tunables.begin(), tunables.end(), [&](auto tunable) { return tunable.IsValid(config); });
+
+    if(it == tunables.end())
+    {
+        MIOPEN_LOG_E("All attempts failed");
+        assert(false);
+    }
+
+    Copy(*it);
+    // std::string kernel_name = GetKernelNameImplicitGemmV4R1Dynamic(*it);
+    // MIOPEN_LOG_I(kernel_name << " : " << ToString());
```

Comment:
Remove or make it at Info2 level, inline `GetKernelNameImplicitGemmV4R1Dynamic()` and uncomment.